src/components/ImageView/index.js

```diff
@@ -45,6 +58,57 @@ class ImageView extends PureComponent {
         document.removeEventListener('mousemove', this.trackMovement.bind(this));
     }
 
+    setImageRegion(width, height, scale) {
+        let x1 = 0;
+        let x2 = 0;
+        let y1 = 0;
+        let y2 = 0;
+        const aspect = width / height;
+        if ((aspect > 1 && this.state.screenWidth / width > 1) || (aspect <= 1 && this.state.screenHeight / height > 1)) {
```

Comment:
I think this `if/else` block can be clarified by extracting some descriptive local variables and adding some comments:

```javascript
const isScreenWiderThanImage = (this.state.screenWidth / width) > 1;
const isScreenTallerThanImage = (this.state.screenHeigh / height) > 1;
if ((aspect > 1 && isScreenWiderThanImage) || (aspect <= 1 && isScreenTallerThanImage)) {
    // Descriptive comment to describe how x1, x2, y1, y2 are set
} else if (aspect > 1 && !isScreenWiderThanImage) {
    // Descriptive comment
} else if (aspect <= 1 && !isScreenTallerThanImage) {
    // Descriptive comment
}
```

Reply:
Ok.